frontend/src/components/shared/FullScreenWrapper/FullScreenWrapper.tsx

```diff
@@ -0,0 +1,141 @@
+/**
+ * @license
+ * Copyright 2018-2019 Streamlit Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { PureComponent } from "react"
+
+import { Button } from "reactstrap"
+import openIconic from "assets/img/open-iconic.svg"
+
+import "./FullScreenWrapper.scss"
+
+export type Size = {
+  height?: number
+  width: number
+}
+
+/*
+ * Function responsible for rendering children.
+ * This function should implement the following signature:
+ * ({ height, width }) => PropTypes.element
+ */
+interface Props {
+  children: (props: Size) => React.ReactNode
+  width: number
+}
+
+interface State {
+  expanded: boolean
+  fullwidth: number
+  fullheight: number
+}
+
+/*
+ * A component that draws a button on the top right of the
+ * wrapper element. OnClick, change the element container
+ * to fixed and cover all screen, updating wrapped element height and width
+ */
+class FullScreenWrapper extends PureComponent<Props, State> {
+  static isFullScreen = false
+  static defaultProps: {
+    Width: 0
+  }
+
+  public state: State = { expanded: false, fullwidth: 0, fullheight: 0 }
+
+  componentDidMount() {
+    this.updateWindowDimensions()
+    window.addEventListener("resize", this.updateWindowDimensions)
+    document.addEventListener("keydown", this.controlKeys, false)
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener("resize", this.updateWindowDimensions)
+    document.removeEventListener("keydown", this.controlKeys, false)
+  }
+
+  controlKeys = (event: any) => {
+    const { expanded } = this.state
+
+    if (event.keyCode === 27 && expanded) {
+      //exit fullscreen
+      this.zoomOut()
+    }
+  }
+
+  zoomIn = () => {
+    document.body.style.overflow = "hidden"
+    FullScreenWrapper.isFullScreen = true
+    this.setState({ expanded: true })
+  }
+
+  zoomOut = () => {
+    document.body.style.overflow = "unset"
+    FullScreenWrapper.isFullScreen = false
+    this.setState({ expanded: false })
+  }
+
+  convertRemToPixels = (rem: number) => {
+    return (
+      rem * parseFloat(getComputedStyle(document.documentElement).fontSize)
+    )
+  }
+
+  updateWindowDimensions = () => {
+    this.setState({
+      fullwidth: window.innerWidth - this.convertRemToPixels(1.5), //0.75rem + 0.75rem
```

Comment:
Where is the 1.5 coming from? If we have it `variables.scss` then you can get it here with `import { SCSS_VARS } from "autogen/scssVariables"`.

Otherwise, at least add a comment explaining where it comes from. And if there is some CSS file that this needs to be in sync with, add a comment in that file pointing to this.

Same below.